### ISSUE TITLE
Can't fall back to telnet.  I've tried several ways any guidance here.

### ISSUE THREAD

**Issue by bkpaine1** [ktbyers/netmiko#2417]
I am trying to do a bunch of show commands from a file, on a bunch of cisco boxes.  The devices, and list of show commands are in separate text files.  SSH works great.  But the client has  many boxes doing telnet (I don't know why) so when ssh fails I try to catch it and fall back to telnet.  I've done this before and it worked but this time I get an error on the telnet attempt with output showing it's still trying to use ssh.

```python
RTR = {
    "device_type": "cisco_ios",
    "ip": "10.10.10.10",
    "username": "user",
    "password": "pass",
}

with open("devices.txt") as DEVICE_LIST:
    for DEVICE in DEVICE_LIST:
        try:
            RTR["device_type"] = "cisco_ios_ssh"
            RTR["ip"] = DEVICE
            print("SSH to device: " + DEVICE)
            RTR["device_type"] = "cisco_ios_ssh"
            net_connect = ConnectHandler(**RTR)
            net_connect.enable()
            with open("showcommands.txt") as COMMANDS:
                output = ""
                for CMD in COMMANDS:
                    DEVICE1 = DEVICE.strip()
                    results = open(DEVICE1 + "-show-commands.txt", "w")
                    output += (
                        CMD + net_connect.send_command(CMD) + "\n" + "+" * 80 + "\n"
                    )
                    results.write(output)
                # print(output)
        except:
            print("Telnet to device: " + DEVICE)
            RTR["device_type"] = "cisco_ios_telnet"
            RTR["port"] = "23"
            net_connect = ConnectHandler(**RTR)
            net_connect.enable()
            with open("showcommands.txt") as COMMANDS:
                output = ""
                for CMD in COMMANDS:
                    DEVICE1 = DEVICE.strip()
                    results = open(DEVICE1 + "-telnet-show-commands.txt", "w")
                    output += (
                        CMD + net_connect.send_command(CMD) + "\n" + "+" * 80 + "\n"
                    )
                    results.write(output)

results.close()
```

I get a bunch of errors it seems to show it still trying ssh.

SSH to device: 172.16.1.203
Telnet to device: 172.16.1.203
---------------------------------------------------------------------------
NoValidConnectionsError                   Traceback (most recent call last)
c:\users\brent\appdata\local\programs\python\python39\lib\site-packages\netmiko\base_connection.py in establish_connection(self, width, height)
    934             try:
--> 935                 self.remote_conn_pre.connect(**ssh_connect_params)
    936             except socket.error as conn_error:

c:\users\brent\appdata\local\programs\python\python39\lib\site-packages\paramiko\client.py in connect(self, hostname, port, username, password, pkey, key_filename, timeout, allow_agent, look_for_keys, compress, sock, gss_auth, gss_kex, gss_deleg_creds, gss_host, banner_timeout, auth_timeout, gss_trust_dns, passphrase, disabled_algorithms)
    367             if len(errors) == len(to_try):
--> 368                 raise NoValidConnectionsError(errors)
    369 

NoValidConnectionsError: [Errno None] Unable to connect to port 22 on 172.16.1.203

During handling of the above exception, another exception occurred:

NetmikoTimeoutException                   Traceback (most recent call last)
<ipython-input-11-37905f63f7e5> in <module>
     26             RTR['device_type'] = 'cisco_ios_ssh'
---> 27             net_connect = ConnectHandler(**RTR)
     28             net_connect.enable()

c:\users\brent\appdata\local\programs\python\python39\lib\site-packages\netmiko\ssh_dispatcher.py in ConnectHandler(*args, **kwargs)
    325     ConnectionClass = ssh_dispatcher(device_type)
--> 326     return ConnectionClass(*args, **kwargs)
    327 

c:\users\brent\appdata\local\programs\python\python39\lib\site-packages\netmiko\cisco\cisco_ios.py in __init__(self, *args, **kwargs)
     16         kwargs.setdefault("_legacy_mode", False)
---> 17         return super().__init__(*args, **kwargs)

**Comment by ktbyers** [comment]
So is this "172.16.1.203" one of the failing devices.

I assume this is a telnet device?


Have you verified you can actually connect to it just using telnet?

**Comment by bkpaine1** [comment]
Yes. I apologize for the loss of python indentation.

I've taken a bunch of your classes and used these skills in my job a lot.
Now I get asked for things like this often lol.

I've built a gns3 lab and .203 is limited to telnet deliberately to
simulate the real world environment it will be working in.

I can putty to it no problem and login with the local credentials. It's
weird. I'm using a skeleton from a similar project I made which works
flawlessly.

The difference is the first project sends config commands whereas this
collects show outputs.  Otherwise they are similar in code structure.  It's
just weird to me. But I can't figure it out.

Brent
